MapleServer2/Servers/Game/FieldManager.cs

```diff
@@ -517,36 +523,61 @@ public FieldObject(int objectId, T value)
             }
         }
 
-        private Task StartHealingSpot()
+        //This is the MapLoop which periodically calls all functions within it
+        //implement all functionalities here, that need to be called periodically and are 
+        //relevant for a given map instance.
+        private Task StartMapLoop()
         {
             return Task.Run(async () =>
             {
-                while (!State.Players.IsEmpty)
+                while (!State.Players.IsEmpty) // test whether this condition is useful, tadeucci said it might be useless.
                 {
-                    foreach (IFieldObject<HealingSpot> healingSpot in State.HealingSpots.Values)
-                    {
-                        CoordS healingCoord = healingSpot.Value.Coord;
-                        foreach (IFieldObject<Player> player in State.Players.Values)
-                        {
-                            if ((healingCoord - player.Coord.ToShort()).Length() < Block.BLOCK_SIZE * 2 && healingCoord.Z == player.Coord.ToShort().Z - 1) // 3x3x1 area
-                            {
-                                int healAmount = (int) (player.Value.Stats[PlayerStatId.Hp].Max * 0.03);
-                                Status status = new Status(new SkillCast(70000018, 1, 0, 1), owner: player.ObjectId, source: healingSpot.ObjectId, duration: 100, stacks: 1);
-
-                                player.Value.Session.Send(BuffPacket.SendBuff(0, status));
-                                BroadcastPacket(SkillDamagePacket.ApplyHeal(status, healAmount));
-
-                                player.Value.Session.Player.Stats.Increase(PlayerStatId.Hp, healAmount);
-                                player.Value.Session.Send(StatPacket.UpdateStats(player, PlayerStatId.Hp));
-                            }
-                        }
-                    }
-
+                    HealingSpot();
+                    MonsterMovement();
                     await Task.Delay(1000);
                 }
             });
         }
 
+        private void MonsterMovement()
+
+        {
+            foreach (IFieldObject<Mob> mob in State.Mobs.Values)
+            {
+                short x = (short) Rand.Next(-70, 70); //random x position, units are block units
+
+                mob.Coord += mob.Value.Speed.ToFloat(); //current position that is given to ControlMob Packet
+                mob.Value.Speed = CoordS.From(x, 0, 0); //speed vector given to ControlMob Packet
+
+                mob.Value.ZRotation = (short) (x * 10); //looking direction of the monster
+
+                //using random animation values, makes it look more lively for now
+                //will be replaced with correct animations on mob creation once animations have been handled. 
+                mob.Value.Animation = (short) Rand.Next(20);
```

Comment:
I noticed there is no packets being sent, is this intended?

Reply:
Yes, the packets are sent within the GetUpdates() function and thereby sync the client with the changes made